### PR TITLE
[expo-updates][android] Add updates database migration transaction helper

### PR DESCRIPTION
**Pull request by wschurman**
# Why

Closes ENG-2628.

https://github.com/expo/expo/pull/15401#discussion_r763635297

# How

Use helper function with block.

# Test Plan

Run all tests and build app.

### CODE REVIEW

File: packages/expo-updates/android/src/main/java/expo/modules/updates/db/UpdatesDatabase.kt [hunk 1]
```diff
@@ -49,56 +49,52 @@ abstract class UpdatesDatabase : RoomDatabase() {
       return instance!!
     }
 
-    val MIGRATION_4_5: Migration = object : Migration(4, 5) {
-      override fun migrate(database: SupportSQLiteDatabase) {
-        // https://www.sqlite.org/lang_altertable.html#otheralter
+    private fun SupportSQLiteDatabase.runInTransactionWithForeignKeysOff(block: SupportSQLiteDatabase.() -> Unit) {
+      // https://www.sqlite.org/lang_altertable.html#otheralter
+      try {
+        execSQL("PRAGMA foreign_keys=OFF")
```
Comment (ide) [inline packages/expo-updates/android/src/main/java/expo/modules/updates/db/UpdatesDatabase.kt:55]: This is a code change, but shouldn't this go outside of the try-block, similar to how beginTransaction() is outside of the nested try-block whose finally-block ends the transaction?

**Review by ide — APPROVED** [review]
TBH I think the point in my inline comment doesn't really matter assuming that turning foreign_keys=ON is idempotent.

File: packages/expo-updates/android/src/main/java/expo/modules/updates/db/UpdatesDatabase.kt (hunk 1 above)
Comment (wschurman) [reply packages/expo-updates/android/src/main/java/expo/modules/updates/db/UpdatesDatabase.kt:55]: @jkhales had a similar comment previously: https://github.com/expo/expo/pull/15401#discussion_r761902404
@esamelson also commented: https://github.com/expo/expo/pull/15401#discussion_r763528313

Comment (wschurman) [reply packages/expo-updates/android/src/main/java/expo/modules/updates/db/UpdatesDatabase.kt:55]: I prefer to leave it as is for now and change them in a follow-up

**Pull request merged**